### PR TITLE
Fixed #28761 -- Documented how an inline formset's prefix works.

### PR DESCRIPTION
**Pull request by JayWelborn**
Docs Improvement for ticket [#28761](https://code.djangoproject.com/ticket/28761).

Existing docs for InlineFormset doesn't mention the default assignment of child model's `related_name` as formset's `prefix.`

### CODE REVIEW

File: docs/topics/forms/modelforms.txt [hunk 1]
```diff
@@ -1230,10 +1230,53 @@ of a model. Here's how you can do that::
 
 Notice how we pass ``instance`` in both the ``POST`` and ``GET`` cases.
 
+Inline formsets and ``related_name``
+------------------------------------
+
+If the foreign key on your child model has a ``related_name`` to the parent,
+this is set to the formset's prefix when using ``inlineformset_factory``.
```
Comment (timmartin) [inline docs/topics/forms/modelforms.txt]: I find this sentence a little confusing - the logic seems backwards. You write "this [the related_name] is set to the formset's prefix", when actually the formset's prefix is the thing being assigned.

I might reword this as "If the foreign key on your child model has a related_name to the parent, the formset uses this name as a prefix."

I'm not sure if it's necessary to specify "when using `inlineformset_factory`", since I think that's implicit in using inline formsets.

File: docs/topics/forms/modelforms.txt [hunk 2]
```diff
@@ -1230,10 +1230,53 @@ of a model. Here's how you can do that::
...
+        title = models.CharField(max_length=100)
+
+    BookFormSet = inlineformset_factory(Author, Book, fields=('title',), extra=1)
+
+``BookFormSet`` above will have its ``prefix`` attribute set to 'books'. This will be added to each
+field's 'name' and 'id' attributes in the template::
+
+    >>> formset = BookFormSet()
+    >>> print(formset)
+    <input type="hidden" name="books-TOTAL_FORMS" value="3" id="id_books-TOTAL_FORMS" /><input type="hidden" name="books-INITIAL_FORMS" value="0" id="id_books-INITIAL_FORMS" /><input type="hidden" name="books-MIN_NUM_FORMS" value="0" id="id_books-MIN_NUM_FORMS" /><input type="hidden" name="books-MAX_NUM_FORMS" value="1000" id="id_books-MAX_NUM_FORMS" />
```
Comment (timmartin) [inline docs/topics/forms/modelforms.txt]: Is there a less messy way to demonstrate that the formset has a prefix? Obviously you could just print the `prefix` member, but that might not get the point across. I looked for other places that the `prefix` element is used and I couldn't see anything suitable.

Actually, the example below of prefixing the data before passing it to the form is probably as good a demonstration as any. I'd be tempted just to print the prefix at this point.

File: docs/topics/forms/modelforms.txt [hunk 3]
```diff
@@ -1230,10 +1230,52 @@ of a model. Here's how you can do that::
...
+
+    class Book(models.Model):
+        author = models.ForeignKey(Author, on_delete=models.CASCADE, related_name='books')
+        title = models.CharField(max_length=100)
+
+    BookFormSet = inlineformset_factory(Author, Book, fields=('title',), extra=1)
+
+``BookFormSet`` above will have its ``prefix`` attribute set to 'books'::
+
+    >>> formset = BookFormSet()
```
Comment (timgraham) [inline docs/topics/forms/modelforms.txt]: I'm not sure that this example is needed. The sentence seems to suffice as I think anyone reading at this point knows what an attribute access looks like. :-)

File: docs/topics/forms/modelforms.txt [hunk 4]
```diff
@@ -1230,10 +1230,52 @@ of a model. Here's how you can do that::
...
+    >>> print(formset.prefix)
+    books
+
+
+The formset's ``prefix`` will be added to each field's 'name' and 'id' attributes when
```
Comment (timgraham) [inline docs/topics/forms/modelforms.txt]: This isn't specific to model formsets, so it may be better to describe the behavior in topics/forms/formsets.html and perhaps link to it from here.

**Comment by JayWelborn** [comment]
Thanks for the review!
I'll get on these changes and update when ready.

**Comment by JayWelborn** [comment]
I've made the example more generic, and added the reference to the `formsets` topic rather than explain prefixes again.

Let me know if there's anything else that needs changing!

File: docs/topics/forms/modelforms.txt [hunk 5]
```diff
@@ -1230,10 +1230,42 @@ of a model. Here's how you can do that::
...
+
+    class Book(models.Model):
+        author = models.ForeignKey(Author, on_delete=models.CASCADE)
+        title = models.CharField(max_length=100)
+
+    BookFormSet = inlineformset_factory(Author, Book, fields=('title',), extra=1)
+
+``BookFormSet`` above will have its ``prefix`` attribute set to 'book_set'. If Book's ForeignKey to Author had a ``related_name``, that would be used instead::
+
+    >>> author = Author.objects.get(name='William Faulkner')
```
Comment (timgraham) [inline docs/topics/forms/modelforms.txt]: I was thinking an example would instead go in the formset docs because this same concern is present there. (The default prefix is 'form').
https://github.com/django/django/blob/6bf85ff7e3b837378589e449ba27be8971d9b14c/django/forms/formsets.py#L263-L265

File: docs/topics/forms/modelforms.txt [hunk 6]
```diff
@@ -1230,10 +1230,42 @@ of a model. Here's how you can do that::
 
 Notice how we pass ``instance`` in both the ``POST`` and ``GET`` cases.
 
+Inline Formsets and ``prefix``
+------------------------------
+
+Inline Formsets are automatically assigned a default ``prefix``. For example, consider the following models and formset::
```
Comment (timgraham) [inline docs/topics/forms/modelforms.txt]: Don't forget to wrap docs at 79 characters.

File: docs/topics/forms/modelforms.txt (hunk 5 above)
Comment (JayWelborn) [reply docs/topics/forms/modelforms.txt]: For regular formsets, the docs already contain examples of passing formsets data with the prefix of `form` and show how to use prefixes in [this section](https://docs.djangoproject.com/en/1.11/topics/forms/formsets/#using-more-than-one-formset-in-a-view).

Do you think they need a separate example explicitly stating that the default prefix is `form`? If so I think either just after or just before [the management form](https://docs.djangoproject.com/en/1.11/topics/forms/formsets/#understanding-the-managementform) explanation would be a good place.

I added this example because InlineFormsets change the default formset prefix in a way that wasn't documented before.

Comment (timgraham) [reply docs/topics/forms/modelforms.txt]: Yea, I guess it's not really clear where that "form-" bit comes from in the formset docs. If that were more clear, then I think we could omit the new examples here and point to the formset docs for examples explaining to substitute the prefix for "form-".

**Comment by JayWelborn** [comment]
Ah I see. That makes sense since that's where the behavior originates.

**Comment by JayWelborn** [comment]
Updated. Created a prefix example in the formsets docs, and used that as the reference link for Inline Formsets

File: docs/topics/forms/modelforms.txt (hunk 5 above)
Comment (JayWelborn) [reply docs/topics/forms/modelforms.txt]: I moved the example to the modelformsets docs, and just put a reference there in the InlineFormsets docs. Hopefully this one will be more clear.

**Comment by JayWelborn** [comment]
This is my first patch. I updated the Authors, and changed the commit message as requested.

Thanks for reviewing!

**Comment by carltongibson** [comment]
@JayWelborn Thanks for you effort. 

@timgraham I think this is good to go.
